src/overlays/actors/ovl_En_Kusa/z_en_kusa.c

```diff
@@ -7,72 +16,485 @@
+// EnKusa_InitCollider
+void func_80A9B630(Actor* thisx, GlobalContext* globalCtx) {
+    EnKusa* this = THIS;
+
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &D_80A9C208);
+    Collider_CylinderUpdate(&this->actor, &this->collider);
+}
+
+void EnKusa_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnKusa* this = THIS;
+
+    Actor_ProcessInitChain(&this->actor, D_80A9C27C);
+    if (globalCtx->csCtx.state != 0) {
+        this->actor.uncullZoneForward += 1000.0f;
+    }
+    // Init for the ColliderCylinder
+    func_80A9B630(thisx, globalCtx);
+
+    func_80061ED4(&this->actor.colChkInfo, NULL, &D_80A9C234);
+    if (this->actor.shape.rot.y == 0) {
+        // Random value I think?
+        s32 rand = (s32)Math_Rand_ZeroFloat(65536.0f);
+        this->actor.posRot.rot.y = (s16)rand;
+        this->actor.initPosRot.rot.y = (s16)rand;
+        this->actor.shape.rot.y = (s16)rand;
+    }
+
+    if (func_80A9AFAC(this, globalCtx, 0.0f) == 0) {
+        Actor_Kill(&this->actor);
+        return;
+    }
+    this->kusaTexObjIndex = Object_GetIndex(&globalCtx->objectCtx, D_80A9C200[thisx->params & 3]);
+
+    if (this->kusaTexObjIndex < 0) {
+        // Prints -> Bank danger!
+        osSyncPrintf("Error : バンク危険！ (arg_data 0x%04x)(%s %d)\n", thisx->params, "../z_en_kusa.c", 0x231);
+        Actor_Kill(&this->actor);
+        return;
+    }
+    // Maybe this is SetupWait?
+    func_80A9B7EC(this);
+}
+
+void EnKusa_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+
+    Collider_DestroyCylinder(globalCtx, &THIS->collider);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kusa/func_80A9B07C.s")
+// SetupWait?
+void func_80A9B7EC(EnKusa* this) {
+    func_80A9AFA0(this, func_80A9B810);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kusa/func_80A9B140.s")
+void func_80A9B810(EnKusa* this, GlobalContext* globalCtx) {
+    if (Object_IsLoaded(&globalCtx->objectCtx, this->kusaTexObjIndex)) {
+        if (this->actor.flags & 0x800) {
+            func_80A9BEAC(this);
+        } else {
+            func_80A9B89C(this);
+        }
+        this->actor.draw = func_80A9C164;
+        this->actor.objBankIndex = this->kusaTexObjIndex;
+        this->actor.flags &= -0x11;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kusa/func_80A9B174.s")
+void func_80A9B89C(EnKusa* this) {
+    func_80A9AFA0(this, func_80A9B8D8);
+    this->actor.flags = this->actor.flags & -0x11;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kusa/func_80A9B1FC.s")
+void func_80A9B8D8(EnKusa* this, GlobalContext* globalCtx) {
+    if (Actor_HasParent(&this->actor, globalCtx) != 0) {
+        func_80A9BA98(this); // Lift up the plant
+        Audio_PlaySoundAtPosition(globalCtx, &this->actor.posRot, 20, NA_SE_PL_PULL_UP_PLANT);
+        return;
+    }
+    // If hit with a sword
+    if ((this->collider.base.acFlags & 2) != 0) {
```

Comment:
```suggestion
    if (this->collider.base.acFlags & 2) {
```
Flag checks like this one can also be treated as boolean.